Nos/Views/RepliesLabel.swift

```diff
@@ -0,0 +1,173 @@
+import Dependencies
+import SwiftUI
+
+struct RepliesLabel: View {
+
+    var repliesDisplayType: RepliesDisplayType
+    var note: Event
+
+    @State private var relaySubscriptions = SubscriptionCancellables()
+    @FetchRequest private var replies: FetchedResults<Event>
+    @Dependency(\.currentUser) var currentUser
+    @EnvironmentObject private var relayService: RelayService
+    @State private var avatars = [URL?]()
+
+    init(repliesDisplayType: RepliesDisplayType, for note: Event) {
+        self.note = note
+        self.repliesDisplayType = repliesDisplayType
+
+        let noteIdentifier = note.identifier ?? ""
+
+        let format = """
+            SUBQUERY(
+                eventReferences,
+                $e,
+                $e.referencedEvent.identifier = %@ AND
+                    ($e.marker = 'reply' OR $e.marker = 'root')
+            ).@count > 0
+        """
+
+        let predicate = NSCompoundPredicate(
+            andPredicateWithSubpredicates: [
+                NSPredicate(format: "kind = 1"),
+                NSPredicate(
+                    format: format,
+                    noteIdentifier,
+                    noteIdentifier
+                ),
+                NSPredicate(format: "deletedOn.@count = 0"),
+                NSPredicate(format: "author.muted = false")
+            ]
+        )
+
+        let fetchRequest = Event.fetchRequest()
+        fetchRequest.includesPendingChanges = false
+        fetchRequest.includesSubentities = false
+        fetchRequest.sortDescriptors = [
+            NSSortDescriptor(keyPath: \Event.identifier, ascending: true)
+        ]
+
+        fetchRequest.predicate = predicate
+        fetchRequest.relationshipKeyPathsForPrefetching = ["author"]
+        _replies = FetchRequest(fetchRequest: fetchRequest)
```

Comment:
I think a lot of this could move out of here, right? And this could be:

```suggestion
        _replies = Event.replies(to: note)
```

I see we already have an `Event.allReplies(to:)` but I'm assuming that's not exactly what you need here. If so, we can use that! If not, I'd just put another class func there.